pkg/asset/manifests/openshift.go

```diff
@@ -271,35 +271,44 @@ func (o *Openshift) Generate(ctx context.Context, dependencies asset.Parents) er
 		}
 		assetData["99_baremetal-provisioning-config.yaml"] = applyTemplateData(baremetalConfig.Files()[0].Data, bmTemplateData)
 	}
-
-	if platform == azuretypes.Name && installConfig.Config.Azure.IsARO() && installConfig.Config.CredentialsMode != types.ManualCredentialsMode {
-		// config is used to created compatible secret to trigger azure cloud
-		// controller config merge behaviour
-		// https://github.com/openshift/origin/blob/90c050f5afb4c52ace82b15e126efe98fa798d88/vendor/k8s.io/legacy-cloud-providers/azure/azure_config.go#L83
-		session, err := installConfig.Azure.Session()
-		if err != nil {
-			return err
-		}
-		config := struct {
-			AADClientID     string `json:"aadClientId" yaml:"aadClientId"`
-			AADClientSecret string `json:"aadClientSecret" yaml:"aadClientSecret"`
-		}{
-			AADClientID:     session.Credentials.ClientID,
-			AADClientSecret: session.Credentials.ClientSecret,
+	if platform == azuretypes.Name {
+		var aro bool
+		client, err := installConfig.Azure.Client()
+		if err == nil {
+			isAro, err := client.CheckIfARO(context.TODO(), installConfig.Config.Azure.ResourceGroupName)
+			if err == nil {
+				aro = isAro
+			}
 		}
+		if aro && installConfig.Config.CredentialsMode != types.ManualCredentialsMode {
+			// config is used to created compatible secret to trigger azure cloud
+			// controller config merge behaviour
+			// https://github.com/openshift/origin/blob/90c050f5afb4c52ace82b15e126efe98fa798d88/vendor/k8s.io/legacy-cloud-providers/azure/azure_config.go#L83
+			session, err := installConfig.Azure.Session()
+			if err != nil {
+				return err
+			}
+			config := struct {
+				AADClientID     string `json:"aadClientId" yaml:"aadClientId"`
+				AADClientSecret string `json:"aadClientSecret" yaml:"aadClientSecret"`
+			}{
+				AADClientID:     session.Credentials.ClientID,
+				AADClientSecret: session.Credentials.ClientSecret,
+			}
```

Comment:
TIL (or remembered)  ARO is generating its own secret for the cloud provider... Does this even work? These are the credentials passed to the installer, but IIUC ARO does not use a secret--they use a certificate. So I would think the value for `ClientSecret` would be empty???

The reason this would not fail during install is because the CCCMO is actually authenticating the cloud provider through its credentials request: https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/manifests/0000_26_cloud-controller-manager-operator_14_credentialsrequest-azure.yaml

So I'm wondering if we can actually just delete this code?

Reply:
@hlipsig Do you know if there's any other specific reason we would need this or should we remove it?

Reply:
I don't think we should resolve this until we get a better sense of whether we can actually remove the secret. https://issues.redhat.com/browse/CORS-3883 is removing the VM identity which is how the installer used to authenticate the cloud provider, but testing has shown that it is no longer needed and the cloud provider seems to be authenticated through cred requests.

Reply:
Per slack conversations we are updating this secret for ARO clusters. I don't think we have any code to create it if it's not already present so I expect it would be breaking for us.

Reply:
> I don't think we have any code to create it if it's not already present so I expect it would be breaking for us.

The CCMO creates a credentials request which it uses to authenticate the cloud provider (see link in my first comment). We have already successfully tested removing the credentials which the installer creates for standalone openshift, so it stands to reason we can do the same for ARO. I think we should test it out so that we can drop this code.

Reply:
Is this resolved?